### PR TITLE
BuildRequestV2 should get outer template as param

### PR DESCRIPTION
**Pull request by rcerven**
Signed-off-by: Robert Cerven <rcerven@redhat.com>

### CODE REVIEW

File: osbs/api.py [hunk 1]
```diff
@@ -362,7 +363,7 @@ def _get_image_stream_info_for_build_request(self, build_request):
         image_stream_tag_name = None
 
         if build_request.has_ist_trigger():
-            image_stream_tag_id = build_request.spec.trigger_imagestreamtag.value
+            image_stream_tag_id = build_json['spec']['triggers'][0]['imageChange']['from']['name']
```
Comment (lcarva) [inline osbs/api.py]: Why this change?

Comment (rcerven) [reply osbs/api.py]: buildrequest2 has empty spec

Comment (lcarva) [reply osbs/api.py]: Good call! Maybe this should be a helper method in BuildRequest? BuildRequestV2 inherits from it so it should be available there too.

File: osbs/build/build_requestv2.py [hunk 2]
```diff
@@ -22,12 +22,13 @@ class BuildRequestV2(BuildRequest):
     Wraps logic for creating build inputs
     """
 
-    def __init__(self, build_json_store, customize_conf=None):
+    def __init__(self, build_json_store, outer_template=None, customize_conf=None):
         """
         :param build_json_store: str, path to directory with JSON build files
```
Comment (sosiouxme) [inline osbs/build/build_requestv2.py]: Can there be a matching `:param` comment?

        :param outer_template: str, path to outer template JSON

Comment (rcerven) [reply osbs/build/build_requestv2.py]: will add

**Comment by lcarva** [comment]
Updated release notes draft:
> - `BuildRequestV2` sets correct outer template for orchestrator and worker builds
